src/Uno.UWP/Security/Authentication/Web/WebAuthenticationBrokerProvider.Android.cs

```diff
@@ -60,62 +65,165 @@ static IEnumerable<string> ExtractSchemes(IntentFilterAttribute a)
+
+			if (!(await StartCustomTabsActivity(url)))
 			{
-				_waitingForCallbackUri = null;
+				// Fall back to opening the system-registered browser if necessary
+				var urlOriginalString = url.OriginalString;
+				var browserIntent = new Intent(Intent.ActionView, global::Android.Net.Uri.Parse(urlOriginalString));
+				WebAuthenticationBrokerRedirectActivity.StartActivity(CurrentActivity!, browserIntent);
 			}
 
-			return result;
+			return await tcsResponse.Task;
 		}
 
-		protected virtual async Task LaunchBrowserCore(WebAuthenticationOptions options,
-			Uri requestUri,
-			Uri callbackUri,
-			CancellationToken ct)
+		private async Task<bool> StartCustomTabsActivity(Uri url)
+		{
+			// Is only set to true if BindServiceAsync succeeds and no exceptions are thrown
+			var success = false;
+			var parentActivity = CurrentActivity;
+
+			var customTabsActivityManager = CustomTabsActivityManager.From(parentActivity);
+			try
+			{
+				if (await BindServiceAsync(customTabsActivityManager))
+				{
+					var customTabsIntent = new CustomTabsIntent.Builder(customTabsActivityManager.Session)
+						.SetShowTitle(true)
+						.Build();
+
+					customTabsIntent.Intent.SetData(global::Android.Net.Uri.Parse(url.OriginalString));
+
+					if (parentActivity?.PackageManager is { } packageManager &&
+						customTabsIntent.Intent.ResolveActivity(packageManager) != null)
+					{
+						WebAuthenticationBrokerRedirectActivity.StartActivity(parentActivity, customTabsIntent.Intent);
+						success = true;
+					}
+				}
+			}
+			finally
+			{
+				try
+				{
+					customTabsActivityManager.Client?.Dispose();
+				}
+				finally
+				{
+				}
+			}
+
+			return success;
+		}
+
+		static Task<bool> BindServiceAsync(CustomTabsActivityManager manager)
+		{
+			var tcs = new TaskCompletionSource<bool>();
+
+			manager.CustomTabsServiceConnected += OnCustomTabsServiceConnected;
+
+			if (!manager.BindService())
+			{
+				manager.CustomTabsServiceConnected -= OnCustomTabsServiceConnected;
+				tcs.TrySetResult(false);
+			}
+
+			return tcs.Task;
+
+			void OnCustomTabsServiceConnected(ComponentName name, CustomTabsClient client)
+			{
+				manager.CustomTabsServiceConnected -= OnCustomTabsServiceConnected;
+				tcs.TrySetResult(true);
+			}
+		}
+
+		internal static bool IsIntentSupported(Intent intent, string? expectedPackageName)
+		{
+			if (Application.Context is not Context ctx || ctx.PackageManager is not PackageManager pm)
+				return false;
+
+			return intent.ResolveActivity(pm) is ComponentName c && c.PackageName == expectedPackageName;
+		}
+
+		internal static bool CanHandleCallback(Uri expectedUrl, Uri callbackUrl)
 		{
-			await Launcher.LaunchUriAsync(requestUri);
+			if (!callbackUrl.Scheme.Equals(expectedUrl.Scheme, StringComparison.OrdinalIgnoreCase))
+				return false;
+
+			if (!string.IsNullOrEmpty(expectedUrl.Host))
+			{
+				if (!callbackUrl.Host.Equals(expectedUrl.Host, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should it be an ignore case comparison?

Reply:
https://www.rfc-editor.org/rfc/rfc3986 - indicates that scheme and host are case in-sensitive where as the rest of the uri is case sensitive.